fixes issue 18 - default behavior conn.send sends to connecting socket

nicholasjstock opened the pull request:
add clients to the mockserver to enable broadcasting.

thoov commented:
@nicholasjstock looks good. I noticed that there are some extra whitespace lines on a few of the files. Can you remove them and then I will merge it in

nicholasjstock commented:
removed the whitespace. 
I then noticed that in the node package WS 
```javascript
wss.on('connection', function(ws) {
 ws.close(); // closes one connection
}
```
only closes one connection, and that to close all connections one does

```javascript
wss.on('connection', function(ws) {
 this.close();
//or wss.close();
}
```
I modified my onclose binding to call the close function of the client rather than the server. 

I also had to modify one of your tests, check it out, see if u agree!

nicholasjstock commented:
@thoov have you had a chance to look at this?

Thanks!

thoov commented on src/client-server-binding.js:
```diff
@@ -0,0 +1,21 @@
+var socketMessageEvent = require('./helpers/message-event');
+var delay = require('./helpers/delay');
+function ClientServerBinding(client, server) {
+		this.send = function(msg) {
```
It looks like the spacing here is off a bit

thoov commented on src/client-server-binding.js:
```diff
@@ -0,0 +1,21 @@
+			if (client.onmessage) {
+				client.onmessage(socketMessageEvent('message', msg, server.url));
+			}
+		}, this)
```
`;`

thoov commented on src/client-server-binding.js:
```diff
@@ -0,0 +1,21 @@
+	}
+
+	this.on = function(type, callback) {
+		server.on.call(server, type, callback)
```
`;`

thoov commented:
@nicholasjstock Overall looks good! Just a few nit picky whitespace things. Can you fix those and rebase then I will merge?
